crates/storage/provider/src/traits/chain.rs

```diff
@@ -0,0 +1,140 @@
+///! Canonical chain state notification trait and types.
+use crate::PostState;
+use auto_impl::auto_impl;
+use core::fmt;
+use reth_primitives::{BlockNumHash, BlockNumber, Receipt, SealedBlockWithSenders, TxHash};
+use std::{collections::BTreeMap, fmt::Formatter, sync::Arc};
+use tokio::sync::broadcast::{Receiver, Sender};
+
+/// Trait that holds blocks and post state of execution those blocks.
+#[auto_impl(&, Arc)]
+pub trait SubChain: Send + Sync {
+    /// Get chain post state.
+    fn state(&self) -> &PostState;
+
+    /// Get chain blocks.
+    fn blocks(&self) -> &BTreeMap<BlockNumber, SealedBlockWithSenders>;
```

Comment:
is it worth it to add a wrapper type + AsRef impls for this, just in case we need helper/convenience functions for this?

Reply:
Removed this trait, and we now have direct access to `Chain`.